osu.Game/Skinning/LegacyBeatmapSkin.cs

```diff
@@ -70,6 +70,14 @@ public override ISample GetSample(ISampleInfo sampleInfo)
             return base.GetSample(sampleInfo);
         }
 
+        protected override DefaultLegacySkin CreateFallbackSkin(IResourceStore<byte[]> storage, IStorageResourceProvider resources)
+        {
+            // for simplicity, beatmap skins don't do lookups on the default skin.
+            // this will mean that fallback always occurs to the user (then default) skin.
+            // this may not offer perfect behaviour, but helps keep things simple.
+            return null;
```

Comment:
It's hard to allow this without changing *all* existing pathways to use the `FindProvider` method. I'm not sure how feasible this is.

A path forward to make this work is to remove this `null` return, and instead override the base implementation of `FindProvider` and block the fallback lookup. The reason I haven't done that is it will cause things like `GetSample` calls (and everything else which isn't aware of `FindProvider`) to incorrectly return the default too early, not falling back to user skins.

I'm interested in hearing suggestions of how we can move forward with this (maybe using `FindProvider` for all `GetDrawable` implementations first and making the "don't fallback" override only in `GetSample` / other methods?).

Having this missing does not regress functionality. It just means the fallback is not available for beatmaps yet - only user skins.

Reply:
What cases would this be handling? Beatmap skin -> default skin fallback in the case that the user skin doesn't provide an asset? Still feels like it should attempt to go through the user skin regardless.

Given my understanding of how stable's skin lookups are done it does sound inevitable that a significant number of (if not all) lookups will eventually need to use `FindProvider()`, though...

Reply:
The issue is that `SkinManager` isn't aware of whether a beatmap skin is requesting legacy resources.

If a beatmap skin provides `hitcircleoverlay`, the `SkinManager` should probably return a `LegacyDefaultSkin` in its pipeline but doesn't right now.